yt_dlp/extractor/eplus.py

```diff
@@ -0,0 +1,99 @@
+from .common import InfoExtractor
+from ..utils import (
+    ExtractorError,
+    try_call,
+    unified_timestamp,
+)
+
+
+class EplusIbIE(InfoExtractor):
+    IE_NAME = 'eplus:inbound'
+    IE_DESC = 'e+ (イープラス) overseas'
+    _VALID_URL = r'https?://live\.eplus\.jp/ex/player\?ib=(?P<id>(?:\w|%2B|%2F){86}%3D%3D)'
+    _TESTS = [{
+        'url': 'https://live.eplus.jp/ex/player?ib=YEFxb3Vyc2Dombnjg7blkrLlrablnJLjgrnjgq%2Fjg7zjg6vjgqLjgqTjg4njg6vlkIzlpb3kvJpgTGllbGxhIQ%3D%3D',
+        'info_dict': {
+            'id': '354502-0001-002',
+            'title': 'LoveLive!Series Presents COUNTDOWN LoveLive! 2021→2022～LIVE with a smile!～【Streaming+(配信)】',
+            'live_status': 'was_live',
+            'upload_date': '20211231',
+            'timestamp': 1640952000,
+            'release_date': '20211231',
+            'release_timestamp': 1640952000,
+            'description': str,
+        },
+        'params': {
+            'skip_download': True,
+            'ignore_no_formats_error': True,
+        },
+        'expected_warnings': [
+            'Could not find the playlist URL. This event may not be accessible',
+            'No video formats found!',
+            'Requested format is not available',
+        ],
+    }]
+
+    def _real_extract(self, url):
+        video_id = self._match_id(url)
+        webpage = self._download_webpage(url, video_id)
+
+        data_json = self._search_json(r'<script>\s*var app\s*=', webpage, 'data json', video_id)
+
+        delivery_status = data_json.get('delivery_status')
+        archive_mode = data_json.get('archive_mode')
+        release_timestamp = try_call(lambda: unified_timestamp(data_json['event_datetime']) - 32400)
+        release_timestamp_str = data_json.get('event_datetime_text')  # JST
+
+        self.write_debug(f'delivery_status = {delivery_status}, archive_mode = {archive_mode}')
+
+        if delivery_status == 'PREPARING':
+            live_status = 'is_upcoming'
+        elif delivery_status == 'STARTED':
+            live_status = 'is_live'
+        elif delivery_status == 'STOPPED':
+            if archive_mode != 'ON':
+                raise ExtractorError(
+                    'This event has ended and there is no archive for this event', expected=True)
+            live_status = 'is_upcoming'
+        elif delivery_status == 'WAIT_CONFIRM_ARCHIVED':
+            live_status = 'is_upcoming'
```

Comment:
Do we want `--wait-for-video` to wait for the VOD? Seems like `post_live` might be more appropriate. Although if we can be sure `--wait-for-video` will not be endless, then it is fine as-is
```suggestion
        elif delivery_status == 'WAIT_CONFIRM_ARCHIVED':
            live_status = 'post_live'
```

Reply:
Thanks for your review.

<br>

> Do we want `--wait-for-video` to wait for the VOD?

Yes, I think we should wait for the VOD.

> Seems like `post_live` might be more appropriate. Although if we can be sure `--wait-for-video` will not be endless, then it is fine as-is

Hmm, am I incorrectly using the "is_upcoming" live status?

<br>

According to the table in my first comment (https://github.com/yt-dlp/yt-dlp/pull/5782#issue-1491140392), the "WAIT_CONFIRM_ARCHIVED" status will eventually switch to "CONFIRMED_ARCHIVE", so the wait won't be infinite.

Reply:
After looking at the core code, I guess it doesn't make much of a difference behavior-wise. IMO `post_live` would be the expected value for a livestream that has ended but hasn't finished processing (this is how `YoutubeIE` does it). `--wait-for-video` will still work with `post_live`

Reply:
Oh, I think I've missed something. We have

```
    live_status:    None (=unknown), 'is_live', 'is_upcoming', 'was_live', 'not_live',
                    or 'post_live' (was live, but VOD is not yet processed)              <---
                    If absent, automatically set from is_live, was_live
```

So, to make the extraction information more accurate, the "live_status" should be "post_live".

@bashonly , thank you for pointing me out.